File: leetcode/0242-valid-anagram/solution.java
class Solution {
    public boolean isAnagram(String s, String t) {
        int[] sChar = new int[26];

        for(char sc : s.toCharArray()){
            sChar[sc - 'a']++;
        }

        for(char tc : t.toCharArray()){
            sChar[tc - 'a']--;
        }

        for(int i = 0; i < 26; i++){
            if(sChar[i] != 0){
                return false;
            }
        }
        return true;
    }
}

File: leetcode/0242-valid-anagram/solution.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t): 
            return False

        char_arr = [0]*26
        base = ord('a')

        for a, b in zip(s, t):
            char_arr[ord(a) - base] += 1
            char_arr[ord(b) - base] -= 1

        return not any(char_arr)
        
